calico-vpp-agent/routing/connectivity/ipip.go

```diff
@@ -26,11 +26,12 @@ import (
 
 type IpipProvider struct {
 	*ConnectivityProviderData
-	ipipIfs map[string]*types.IPIPTunnel
+	ipipIfs    map[string]*types.IPIPTunnel
+	ipipRoutes map[uint32][]*types.Route
```

Comment:
We should be able to keep only a refcount of routes per tunnel ?
That would simplify the `DelConnectivity` logic

Also we should clear this state when VPP restarts (in `OnVppRestart()`)
And rescan the current state in vpp when we (the agent) restarts `RescanState()`

Reply:
Re-thinking about this, maybe it's worth keeping a map of the routes (just in case BGP calls us twice with the same route).
We can add the route to VPP only if it's not already present in the map
This will guarantee that on restart we keep the map intact.

So something like `map[uint32]map[string]bool`